integration-tests/modules/__tests__/product/workflows/product-options-removal-validation.spec.ts

```diff
@@ -0,0 +1,365 @@
+import { createAndLinkProductOptionsToProductWorkflow } from "@medusajs/core-flows"
+import { medusaIntegrationTestRunner } from "@medusajs/test-utils"
+import { IProductModuleService } from "@medusajs/types"
+import { MedusaError, Modules } from "@medusajs/utils"
+
+jest.setTimeout(50000)
+
+medusaIntegrationTestRunner({
+  env: {},
+  testSuite: ({ getContainer }) => {
+    describe("Product Options Removal Validation", () => {
+      let appContainer
+      let productModule: IProductModuleService
+
+      beforeAll(async () => {
+        appContainer = getContainer()
+        productModule = appContainer.resolve(Modules.PRODUCT)
+      })
+
+      describe("removeProductOptionFromProduct", () => {
+        it("should throw error when trying to remove ProductOption used by variant", async () => {
+          // Create product with option
```

Comment:
nit: those comments can be removed, the actions are pretty self explanatory in themselves :D